packages/esm-reports-app/src/components/reports-data-overview.component.tsx

```diff
@@ -0,0 +1,119 @@
+import dayjs from 'dayjs';
+
+const ReportsDataOverviewComponent: React.FC = () => {
+  const { t } = useTranslation();
+  const [selectedReport, setSelectedReport] = useState('');
+  const [reportParameters, setReportParameters] = useState<Record<string, string>>({});
+  const [isLoading, setIsLoading] = useState(false);
+  const [reportData, setReportData] = useState(null);
+
+  const { reportDefinitions } = useReportDefinitions();
+
+  const selectedReportDefinition = useMemo(() => {
+    return reportDefinitions?.find((report) => report.uuid === selectedReport);
+  }, [selectedReport, reportDefinitions]);
+
+  const handleParameterChange = (parameterName: string, value: string) => {
+    const formattedDate = value ? dayjs(value).format('YYYY-MM-DDTHH:mm:ss.SSSZ') : '';
+    setReportParameters((prev) => ({
+      ...prev,
+      [parameterName]: formattedDate,
+    }));
+  };
+
+  const handleFetchReport = () => {
+    if (!selectedReport) {
+      showSnackbar({
+        kind: 'error',
+        title: t('error', 'Error'),
+        subtitle: t('selectReportFirst', 'Please select a report first'),
+      });
+      return;
+    }
+
+    setIsLoading(true);
+    getReportDataObservable(selectedReport, reportParameters)
+      .pipe(take(1))
+      .subscribe(
+        (response) => {
+          setIsLoading(false);
+          setReportData(response.data);
+          showSnackbar({
+            kind: 'success',
+            title: t('reportFetched', 'Report fetched successfully'),
+          });
+        },
+        (error) => {
+          setIsLoading(false);
+          showSnackbar({
+            kind: 'error',
+            title: t('error', 'Error'),
+            subtitle: error?.message || t('errorFetchingReport', 'Error fetching report'),
+          });
+        },
+      );
+  };
+
+  return (
+    <div>
+      <ExtensionSlot name="breadcrumbs-slot" className={styles.breadcrumb} />
+      <div className={styles.mainPanelDiv}>
+        <div className={styles.reportsLabelDiv}>
+          <h3>{t('reports', 'Reports')}</h3>
+        </div>
+        <div className={styles.filterForm}>
+          <div className={styles.formField}>
+            <Select
+              id="report-select"
+              labelText={t('selectReport', 'Select Report')}
+              value={selectedReport}
+              onChange={(e) => {
+                setSelectedReport(e.target.value);
+                setReportParameters({});
+                setReportData(null);
+              }}
+            >
+              <SelectItem text={t('selectReport', 'Select Report')} value="" />
+              {reportDefinitions?.map((report) => (
+                <SelectItem key={report.uuid} text={report.name} value={report.uuid} />
+              ))}
+            </Select>
+          </div>
+          {selectedReportDefinition?.parameters?.map((parameter) => (
+            <div key={parameter.name} className={styles.formField}>
+              <DatePicker
```

Comment:
While this is potentially screwy, we have an `OpenmrsDatePicker` component that should be used here.